src/hal/alarm.cpp

```diff
@@ -0,0 +1,76 @@
+
+#include <stdbool.h>
+
+#include "../hal/hal.h"
+#include "../hal/alarm.h"
+#include "../hal/timer.h"
+
+#include <Arduino.h>
+
+int alarmHalInit(void)
+{
+  pinMode(ALARM_PIN, OUTPUT);
+}
+
+int alarmHalRing(unsigned int pattern)
+{
+  static unsigned int currentPattern = ALARM_HAL_OFF;
+  static bool alarmOn = false;
+  static struct timer timer;
+  uint32_t alarmTime;
+
+  // Check to see if new tone needs to be played
+  if (pattern != currentPattern) {
```

Comment:
Should we check that `pattern` is either 0-4?

Reply:
Done in latest commits